backend/src/main/java/com/benine/backend/video/MJPEGStreamReader.java

```diff
@@ -0,0 +1,193 @@
+package com.benine.backend.video;
+
+import java.awt.image.BufferedImage;
+import java.io.BufferedInputStream;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.imageio.ImageIO;
+
+/**
+ * StreamReader for Motion JPEG streams.
+ */
+public class MJPEGStreamReader extends StreamReader {
+
+  private BufferedInputStream bufferedStream;
+  private byte[] snapShot;
+
+  /**
+   * Creates a new MJPEGStreamReader.
+   *
+   * @param url The url fo the stream.
+   * @throws IOException if the inputstream cannot be read.
+   */
+  public MJPEGStreamReader(String url) throws IOException {
+      this(new Stream(url));
+  }
+
+  /**
+   * Creates a new MJPEGStreamReader.
+   *
+   * @param stream A stream object.
+   * @throws IOException If the inputstream cannot be read.
+   */
+  public MJPEGStreamReader(Stream stream) {
+    this.bufferedStream = new BufferedInputStream(stream.getInputStream());
+    processStream();
+  }
+
+  @Override
+  public void run() {
+    while (!Thread.interrupted()) {
+      processStream();
+    }
+  }
+
+  /**
+   * Processes a stream by fetching an image
+   * from the stream and updating the snapshot if possible.
+   */
+  public void processStream() {
+    try {
+      snapShot = getImage();
+
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Peeks into the next amount of bytes and returns them.
+   *
+   * @param amount The amount of bytes to check.
+   * @return An int[] array containing the checked bytes in order.
+   * @throws IOException when the next bytes cannot be read from the stream.
+   */
+  private int[] checkNextBytes(int amount) throws IOException {
+    int[] byteArray = new int[amount];
+
+    bufferedStream.mark(amount);
+    for (int i = 0; i < amount; i++) {
+      byteArray[i] = bufferedStream.read();
```

Comment:
I've been thinking about that, but reading from an input stream consumes the input, so looking for the header by reading consumes the first 2 bytes of the JPEG image (the flag). I think it is more clear to "peek" into the next 2 bytes instead of reading them, because if they are read they need to be re-added in front of the JPEG image.